HomeTask_9.py

```diff
@@ -1,10 +1,29 @@
+"""HomeTask_9.
+
+write script that will read "input.txt" file and find there all characters from english alphabet
+collect these characters and their positions in file
+after info collected this info as text write to "output.txt" file in such format
+####################
+<first found letter> -> pos1
+<next_letter> -> pos2
+<next_letter -> pos3
+etc
+#######################
+"""
 import re
-with open('input.txt', 'r') as inp:
+
+with open('input.txt', 'r') as inp:  # open the file for reading
     data = inp.read()
-    regexp = re.findall(r'[a-zA-Z]', data)
-    with open('output.txt', 'w') as out:
-        for i in regexp:
-            out.write(f'{i} -> pos{regexp.index(i) + 1}\n')
-    with open('output.txt', 'r') as out:
-        output_text = out.read()
-print(output_text)
+    regexp = re.finditer(r'[a-zA-Z]', data)  # find the all characters from english alphabet
+    d1 = {i.group(): i.start() for i in regexp}  # collect these characters and their positions in file
```

Comment:
Not correct structure for storing results. Cause letter could have duplicates and this duplicated and their positions will be missed. As you remember dict does not allow to store the same keys. You can add at the beginning of input file "e" letter and check with your code what I have mentioned above ^^